### PR TITLE
Adds UART Bus

### PR DESCRIPTION
**Pull request by brentru**
This pull request adds basic message commands for the upcoming uart feature.

Addresses https://github.com/adafruit/Wippersnapper_Protobuf/issues/128

### CODE REVIEW

**Comment by brentru** [comment]
Also updates workflow to remove the nodeJS 12 depreciation warning and adds a downloadable `wippersnapper/` folder for insertion into `WS_Arduino/src` builds

File: proto/wippersnapper/uart/v1/uart.proto [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import "wippersnapper/i2c/v1/i2c.proto";
+
+/**
+* UARTBusData represents a message to configure a UART bus for communication with a device.
+* NOTE: This message is never sent directly, it is packed inside UARTDeviceAttachRequest.
+*/
+message UARTBusData {
+  int32 baudrate        = 1; /** The baudrate to use for UART communication (may be a common baud rate such as: 1200bps, 2400bps, 4800bps, 19200bps, 38400bps, 57600bps, or 115200bps). */
+  int32 pin_rx         = 2; /** The pin on which to receive UART stream data. */
+  int32 pin_tx         = 3; /** The pin on which to transmit UART stream data. */
```
Comment (lorennorman) [inline proto/wippersnapper/uart/v1/uart.proto:16]: Comparing with `pin.proto`, why is this one using `int32` for the pin when others use the `string` name?

File: proto/wippersnapper/uart/v1/uart.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+
+# uart.proto
+
+This file details the WipperSnapper messaging API for interfacing with a UART bus.
+
+## WipperSnapper Components
+
+The following WipperSnapper components utilize `pwm.proto`:
```
[the remark this answers is not in the document]
Comment (brentru) [reply proto/wippersnapper/uart/v1/uart.md:8]: Yes it is, correcting in next commit

File: proto/wippersnapper/uart/v1/uart.proto (hunk 1 above)
Comment (brentru) [reply proto/wippersnapper/uart/v1/uart.proto]: It should use the `string` name, corrected in next commit

File: proto/wippersnapper/uart/v1/uart.proto [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  bool is_success  = 2; /** True if the UARTInitRequest was successful, False otherwise. */
+}
+
+/**
+* UARTDeviceEvent represents incoming data from a UART sensor.
+*
+message UARTDeviceEvent {
+  string device_id = 1[(nanopb).max_size = 15];                    /** The unique identifier of the device to attach to the UART bus, from Adafruit_WipperSnapper_Components. */
+  repeated wippersnapper.i2c.v1.SensorEvent sensor_event  = 2[(nanopb).max_count = 15]; /** A, optionally repeated, SensorEvent from a sensor. */
+}
```
Comment (lorennorman) [inline proto/wippersnapper/uart/v1/uart.proto:45]: What's up here, is this commented out on purpose? If so, Line 44 accidentally breaks out of the doc-block due the line above closing itself with `*/`, and it's still being referenced in `signal.proto`.

**Review by lorennorman — COMMENTED** [review]
This is still incomplete, I think?

Still loving the mermaid diagrams, and good work on making the pins strings.

**Comment by brentru** [comment]
@lorennorman  It's not incomplete, it doesn't have GPS yet but should work for the PM UART sensors. GPS will be a new PR.

**Comment by lorennorman** [comment]
@brentru Ok, I see. Are you going to cover `UARTDeviceEvent` in the mermaid diagram?

**Comment by brentru** [comment]
@lorennorman  Added to my list, will ping you when done

**Comment by brentru** [comment]
@lorennorman  Covered `UARTDeviceEvent` message within the mermaid diagram, this PR is ready.

**Review by lorennorman — APPROVED** [review]

**Pull request merged**